v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/transforms/EventMetadata.java

```diff
@@ -0,0 +1,190 @@
+        }
+
+        agent = new Agent();
+        agent.name = "";
+        agent.version = StringUtils.isNotBlank(pubSubToElasticsearchOptions.getElasticsearchTemplateVersion()) ?
+                pubSubToElasticsearchOptions.getElasticsearchTemplateVersion()
+                : "1.0.0";
+        agent.id = "";
+
+        dataStream.dataset = pubSubToElasticsearchOptions.getDataset();
+        dataStream.namespace = pubSubToElasticsearchOptions.getNamespace();
+
+        service.type = event.dataset = pubSubToElasticsearchOptions.getDataset();
+    }
+
+    public List<JsonNode> asList() {
+        JsonNode root = objectMapper.convertValue(this, JsonNode.class);
+
+        List<JsonNode> nodes = new ArrayList<>();
+
+        for (JsonNode node : root) {
+            nodes.add(node);
+        }
+
+        return nodes;
+    }
+
+    private class Agent {
+        @JsonProperty("type")
+        private final String type = "dataflow";
+        @JsonProperty("name")
+        private String name;
+        @JsonProperty("version")
+        private String version;
+        @JsonProperty("id")
+        private String id;
+    }
+
+    private class DataStream {
+        @JsonProperty("type")
+        private final String type = "logs";
+        @JsonProperty("dataset")
+        private Dataset dataset;
+        @JsonProperty("namespace")
+        private String namespace;
+    }
+
+    private class Ecs {
+        @JsonProperty("version")
+        private final String version = "1.10.0";
+    }
+
+    private class Service {
+        @JsonProperty("type")
+        private Dataset type;
+    }
+
+    private class Event {
+        @JsonProperty("module")
+        private final String module = "gcp";
+        @JsonProperty("dataset")
+        private Dataset dataset;
+    }
+
+    public static EventMetadata build(String inputMessage, PubSubToElasticsearchOptions pubSubToElasticsearchOptions) {
+        EventMetadata eventMetadata = new EventMetadata(inputMessage, pubSubToElasticsearchOptions);
+        eventMetadata.enrich();
+        return eventMetadata;
+    }
+
+    private void enrich() {
+        try {
+            enrichedMessage = objectMapper.readTree(inputMessage);
+            ObjectNode metadata = (ObjectNode) objectMapper.readTree(
+                    objectMapper.writerFor(EventMetadata.class).writeValueAsString(this));
+            ((ObjectNode) enrichedMessage).putAll(metadata);
+            ((ObjectNode) enrichedMessage).remove("timestamp");
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Error processing input message.", e);
+        }
+    }
+
+    public String getEnrichedMessageAsString() {
+        try {
+            return objectMapper.writeValueAsString(enrichedMessage);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException("Error building enriched message.", e);
```

Comment:
Include the value that is causing exception in all error messages. For e.g:
```
throw new IllegalStateException("Exception occurred while building enriched message {} : {} ", enrichedMessage , e);
```

Use similar format for all exception messages

Reply:
This format of message is compatible with logs, but not for exception message.